php/libraries/Visit.class.inc

```diff
@@ -0,0 +1,220 @@
+ *
+ * @category Main
+ * @package  Main
+ * @author   Mélanie Legault <melanie.legault2@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+class Visit
+{
+    /**
+     * Return an array containing the label and legacyLabel of a visit
+     *
+     * @param integer $id the identifier of the label
+     *
+     * @throws DatabaseException
+     *
+     * @return array a hash with both label
+     *                 hash is of the form
+     *                 array('Label'       => 'label',
+     *                       'LegacyLabel' => 'legacy label')
+     */
+    public function getLabel($id)
+    {
+        $DB = Database::singleton();
+
+        $result = $DB->pselect(
+            "SELECT VisitLabel, VisitLegacyLabel
+             FROM visit
+             WHERE VisitID = :id",
+            array('id' => $id)
+        );
+        return $result;
+    }
+
+    /**
+     * Return an array of hash containing all the visit available
+     * for a project
+     *
+     * @param array $param array['projectID'] the ID of the project
+     *                     array['subprojectID'] the ID of the subproject
+     *                     for which the visit should be available
+     *                     if none, all visit
+     *
+     * @throws DatabaseException
+     *
+     * @return array an array of hash with id and both label
+     *               array is of the form
+     *               array(0 => array('VisitID'     => visit ID,
+     *                                'Label'       => 'label',
+     *                                'LegacyLabel' => 'legacy label'))
+     */
+    public static function getList($param = array())
+    {
+        $DB = Database::singleton();
+
+        $where      = "";
+        $queryParam = array();
+        if (isset($param['projectID']) && !is_null($param['projectID'])) {
+            $where = " JOIN visit_project_subproject_rel using(VisitID)
+                       WHERE ProjectID = :projectID";
+            $queryParam['projectID'] = $param['projectID'];
+            if (empty($param['subprojectID'])) {
+                $where .= " AND SubprojectID IS NULL";
+            } else {
+                $where .= " AND SubprojectID = :subprojectID";
+                $queryParam['subprojectID'] = $param['subprojectID'];
+
+            }
+        } elseif (isset($param['subprojectID'])
+            && !is_null($param['subprojectID'])
+        ) {
+            $where = " JOIN visit_project_subproject_rel using(VisitID)
+                       WHERE SubprojectID = :subprojectID AND ProjectID IS NULL";
+            $queryParam['subprojectID'] = $param['subprojectID'];
+        }
+        $result = $DB->pselect(
+            "SELECT VisitID, VisitLabel, VisitLegacyLabel
+             FROM visit" . $where,
+            $queryParam
+        );
+        return $result;
+    }
+
+    /**
+     * Modify the label and legacyLabel of a visit
```

Comment:
As a model, this would be better off as an immutable function returning a new Visit, as a database modifying object, this should document that it modifies the database (as opposed to modifying the class.)

Reply:
I have some questions about that to bring up

Reply:
..so are you going to ask them?